lib/chef/exceptions.rb

```diff
@@ -97,6 +97,7 @@ class GroupIDNotFound < ArgumentError; end
     class ConflictingMembersInGroup < ArgumentError; end
     class InvalidResourceReference < RuntimeError; end
     class ResourceNotFound < RuntimeError; end
+    class ProviderNotFound < ArgumentError; end
```

Comment:
@jdm fixed. I think RuntimeError makes marginally more sense. And I like it when names are consistent.